crates/common/src/ether/evm/core/types.rs

```diff
@@ -304,7 +305,7 @@ pub fn find_cast(line: &str) -> Result<(Range<usize>, String), Error> {
 
             // find where the cast ends
             let range = find_balanced_encapsulator(&line[end..], ('(', ')'))?;
-            Ok((end + range.start..end + range.end, cast_type))
+            Ok((end + range.start..end + range.end, to_type(&cast_type)))
```

Comment:
i wonder if we should replace `to_type` impl with `FromStr`

Reply:
I believe `ParamType` doesn't implement `FromStr`. It would be way cleaner if it did

Reply:
gotcha